Tella/Data/Database/TellaDataBase.swift

```diff
@@ -509,24 +509,36 @@ class TellaDataBase {
         
         guard let reportId, let report = self.getReport(reportId: reportId) else { return 0}
         
-        deleteReportFiles(report: report)
-        
+        deleteReportFiles(reportIds: [reportId])
+
         let reportCondition = [KeyValue(key: D.cReportId, value: report.id as Any)]
         return try statementBuilder.delete(tableName: D.tReport,
                                            primarykeyValue: reportCondition)
     }
-    
-    func deleteReportFiles(report:Report) {
+
+    func deleteSubmittedReport() throws -> Int {
```

Comment:
Maybe instead of throwing an error from the function, we can catch the error within the function and handle it appropriately, with a `try catch`?